### ISSUE TITLE
Cannot have a global last-built-version link

### ISSUE THREAD

**Issue by vanrein** [arpa2/mkhere#18]
https://github.com/arpa2/mkhere/commit/c5a76bf8c309d16fa286d1953059ebcfb5e4329d#diff-fd7f63d3424a7021111c10fa9bbbd2aaR402

This is not quite right; it might cause one client to modify  what another client sees, especially during parallel build processes.

Maybe we can identify clients, and give each their own links space?

**Comment by vanrein** [comment]
A default directory is possible, but only in a certain condition.

  * When the global source tree (of recipes) builds with the default `VERSION_pkg` and `VARIANT_pkg`
  * These variables can be used in dependencies even when they are undefined
  * While building `pkg` and its `VERSION_pkg` is undefined, its `default_VERSION` takes over.  A link can be used to allow the default to be used in later builds that also miss `VERSION_pkg`.
  * While building `pkg` and its `VARIANT_pkg` is undefined, its `default_VARIANT` takes over.  A link can be used to allow the default to be used in later builds that also miss `VARIANT_pkg`.
  * Basically, the link location is the before-defaults-are-set `LNK_TREE`, and the target is the after-defaults-are-set `DIR_TREE`.

Now, why didn't I think of that right away... it being so trivial :-S

**Comment by vanrein** [comment]
Resolved in 7c81e1bb5a89b615eec514ed5b66c2e8d4454b00 in `master`.
